content/en/tracing/manual_instrumentation/ruby.md

```diff
@@ -1,23 +1,130 @@
 further_reading:
-    - link: 'tracing/guide/instrument_custom_method'
-      tag: 'Guide'
-      text: 'Instrument a custom method to get deep visibility into your business logic'
     - link: 'tracing/connect_logs_and_traces'
       tag: 'Documentation'
       text: 'Connect your Logs and Traces together'
-    - link: 'tracing/opentracing'
-      tag: 'Documentation'
-      text: 'Implement Opentracing across your applications'
     - link: 'tracing/visualization/'
       tag: 'Documentation'
       text: 'Explore your services, resources, and traces'
 ---
+<div class="alert alert-info">
+If you have not yet read the instructions for auto-instrumentation and setup, please start with the <a href="https://docs.datadoghq.com/tracing/setup/ruby/">Ruby Setup Instructions</a>.
+</div>
+
+This page details common use cases for adding and customizing observability with Datadog APM.
+
+## Adding Tags
+
+Add custom [span tags][1] to your [spans][2] to customize your observability within Datadog.  The span tags are applied to your incoming traces, allowing you to correlate observed behavior with code-level information such as merchant tier, checkout amount, or user ID.
 
-If you aren't using supported library instrumentation (see [library compatibility][1]), you may want to to manually instrument your code. Adding tracing to your code is easy using the `Datadog.tracer.trace` method, which you can wrap around any Ruby code:
+### Add custom span tags
+
+Add custom tags to your spans corresponding to any dynamic value within your application code such as `customer.id`.
+
+{{< tabs >}}
+{{% tab "Active Span" %}}
+Access the current active [span][1] from any method within your code. **Note**: If the method is called and there is no span currently active, `active_span` is `nil`.
+
+```ruby
+require 'ddtrace'
+
+# get '/shopping_cart/:customer_id', to: 'shopping_cart#index'
+class ShoppingCartController < ApplicationController
+  # GET /shopping_cart
+  def index
+    # Get the active span
+    current_span = Datadog.tracer.active_span
+    # customer_id -> 254889
+    current_span.set_tag('customer.id', params.permit([:customer_id])) unless current_span.nil?
+
+    # [...]
+  end
+
+  # POST /shopping_cart
+  def create
+    # [...]
+  end
+end
+```
+
+[1]: /tracing/visualization/#spans
+{{% /tab %}}
+
+{{% tab "Manually Instrumented Spans" %}}
+
+Add [tags][1] directly to `Datadog::Span` objects by calling `#set_tag`:
+
+```ruby
+# An example of a Sinatra endpoint,
+# with Datadog tracing around the request.
+get '/posts' do
+  Datadog.tracer.trace('web.request') do |span|
+    span.set_tag('http.url', request.path)
+    span.set_tag('<TAG_KEY>', '<TAG_VALUE>')
+  end
+end
+```
+
+
+[1]: /tracing/visualization/#span-tags
+{{% /tab %}}
+{{< /tabs >}}
+
+### Adding tags globally to all spans
+
+Add [tags][1] to all [spans][2] by configuring the tracer with the `tags` option:
+
+```ruby
+Datadog.configure do |c|
+  c.tracer tags: { 'env' => 'dev' }
```

Comment:
This works but is deprecated; keep in mind this will be changing in the future to:

```
Datadog.configure do |c|
  c.tags = { 'team' => 'qa' }
end
```

Reply:
Can we change the snippet now to the snippet you provided? Don't want to be recommending a deprecated way of doing this.